engine/execution/ingestion/core.go

```diff
@@ -506,15 +512,37 @@ func (e *Core) execute(ctx context.Context, executable *entity.ExecutableBlock)
 }
 
 func (e *Core) fetch(missingColls []*block_queue.MissingCollection) error {
+	missingCount := 0
 	for _, col := range missingColls {
-		err := e.collectionFetcher.FetchCollection(col.BlockID, col.Height, col.Guarantee)
+
+		// if we've requested this collection, we will store it in the storage,
```

Comment:
[The same logic](https://github.com/onflow/flow-go/blob/leo/check-local-collection-if-exist/engine/execution/ingestion/engine.go#L911-L927) exists in the old ingestion engine